build_script.py

```diff
@@ -81,11 +81,17 @@ def main():
                         action="store_true",
                         dest="test",
                         default=False)
+    parser.add_argument("--arch",
+                        help="target architecture",
+                        action="store",
+                        dest="arch",
+                        default=None)
     args = parser.parse_args()
 
     swiftc = os.path.abspath(args.swiftc)
     build_dir = os.path.abspath(args.build_dir)
     swift_build_dir = os.path.abspath(args.swift_build_dir)
+    arch = args.arch
```

Comment:
nit-pick: You could simply reference `args.arch` below, and eliminate this line. Other parameters, such as `args.build_style`, follow that pattern.